### PR TITLE
refactor: single endpoint for all queries

### PR DESCRIPTION
**Pull request by MattScarthSaunders**

### CODE REVIEW

File: protocol/src/domains/SqlDomain.ts [hunk 1]
```diff
@@ -1,30 +1,19 @@
...
         connectionUrl: string;
     }): Promise<{}>;
 
-    executeDefinition(req: {
+    query(req: {
         connectionUrl: string;
-        definition: string;
+        query: string;
+        params?: any[];
     }): Promise<{}>;
```
Comment (inca) [inline protocol/src/domains/SqlDomain.ts]: Should still return something right? Weird that TS didn't catch that 🤔

File: src/main/connections/postgres/PostgresConnection.ts [hunk 2]
```diff
@@ -16,45 +16,39 @@ export class PostgresConnection extends BaseConnection {
...
-    release() {
-        this.client.release();
-    }
-
-    private async execute(text: string, params?: any[]) {
         try {
-            return await this.client.query(text, params);
+            const res = await this.client.query(text, params);
+
+            if (res.rows.length && res.rowCount) {
```
Comment (inca) [inline src/main/connections/postgres/PostgresConnection.ts:23]: Does it always have `res.rows`? (just curious as I don't know what the result type is)

**Review by inca — APPROVED** [review]
Looks good 👍  Except for one minor oddity in protocol

File: src/main/connections/postgres/PostgresConnection.ts (hunk 2 above)
Comment (MattScarthSaunders) [reply src/main/connections/postgres/PostgresConnection.ts:23]: yeah, with `pg` the return type of the client is consistent, so in the case of DDL queries it's always an empty array, a rowCount of 0 and empty field data. So figured theres just no point in returning anything in that case.

`mysql2` is much more particular/specific about typing and return types which is why the code is a bit more involved in that one

File: protocol/src/domains/SqlDomain.ts (hunk 1 above)
Comment (MattScarthSaunders) [reply protocol/src/domains/SqlDomain.ts]: Huh, yeah that's weird - oops! Thanks for catching it

**Pull request merged**